src/_P042_Candle.ino

```diff
@@ -76,422 +83,440 @@ enum ColorType {
-                if (Candle_type == 2) {
-                  type_Simple_Candle();
-                }
-                if (Candle_type == 3) {
-                  type_Advanced_Candle();
-                }
-                Candle_Update = millis() + random(25, 150);
-              }
-              break;
-            }
+        case 1: // Update for LIGHT
+        {
+          type_Static_Light();
+          break;
+        }
 
-          case 4:   // Update for Police
-            {
-              if (timeOutReached(Candle_Update)) {
-                type_Police();
-                Candle_Update = millis() + 150;
-              }
-              break;
+        case 2: // Random Updates for Simple Candle, Advanced Candle, Fire Simulation
+        case 3:
+        {
+          if (timeOutReached(Candle_Update)) {
+            if (Candle_type == 2) {
+              type_Simple_Candle();
             }
 
-          case 5:   // Update for Blink
-            {
-              if (timeOutReached(Candle_Update)) {
-                type_BlinkStrobe();
-                Candle_Update = millis() + 100;
-              }
-              break;
+            if (Candle_type == 3) {
+              type_Advanced_Candle();
             }
+            Candle_Update = millis() + random(25, 150);
+          }
+          break;
+        }
 
-          case 6:   // Update for Strobe
-            {
-              type_BlinkStrobe();
-              break;
-            }
-          case 7:   // Update for ColorFader
-            {
-              if (timeOutReached(Candle_Update)) {
-                type_ColorFader();
-                Candle_Update = millis() + 2000;
-              }
-              break;
-            }
+        case 4: // Update for Police
+        {
+          if (timeOutReached(Candle_Update)) {
+            type_Police();
+            Candle_Update = millis() + 150;
+          }
+          break;
         }
 
-        Candle_pixels->show();
+        case 5: // Update for Blink
+        {
+          if (timeOutReached(Candle_Update)) {
+            type_BlinkStrobe();
+            Candle_Update = millis() + 100;
+          }
+          break;
+        }
 
-        success = true;
-        break;
+        case 6: // Update for Strobe
+        {
+          type_BlinkStrobe();
+          break;
+        }
+        case 7: // Update for ColorFader
+        {
+          if (timeOutReached(Candle_Update)) {
+            type_ColorFader();
+            Candle_Update = millis() + 2000;
+          }
+          break;
+        }
       }
 
+      Candle_pixels->show();
+
+      success = true;
+      break;
+    }
+
     case PLUGIN_READ:
-      {
-        UserVar[event->BaseVarIndex] = Candle_red * 65536 + Candle_green * 256 + Candle_blue;
-        UserVar[event->BaseVarIndex + 1] = Candle_bright;
-        UserVar[event->BaseVarIndex + 2] = Candle_type;
+    {
+      UserVar[event->BaseVarIndex]     = Candle_red * 65536 + Candle_green * 256 + Candle_blue;
```

Comment:
isn't there some variant for 24 bit like `makeWord`?

Reply:
No, but improved by using shifts.